### PR TITLE
[BF-58] FEAT: Implement help page logic

### PR DESCRIPTION
**Pull request by alexkocharyan01**

### CODE REVIEW

File: src/pages/profile/help/index.tsx [hunk 1]
```diff
@@ -1,22 +1,83 @@
...
-    <Input placeholder="Headline" />
-    <Input
-      placeholder="Type your problem here..."
-      color="primary"
-      rows={7}
-    />
-    <Button>Submit</Button>
-  </div>
-)
+import { helpValidationSchema } from '../../../validation/support/help'
```
Comment (ArthurO2002) [inline src/pages/profile/help/index.tsx:10]: Add validation to tsconfig to have
```suggestion
import { helpValidationSchema } from '@validation/support/help'
```

File: src/pages/profile/help/index.tsx [hunk 2]
```diff
@@ -1,22 +1,83 @@
...
+        problem: data.problem,
+      })
+    )
+  }
+
+  return (
+    <div className="grid gap-4 rounded bg-gray-thin p-5 xl:p-10">
+      <h2 className="text-lg">Tell us about your problem:</h2>
+      <p className="text-base text-black-light">
+        Try to support your explanation with a screenshot, with that we can solve your problem
```
Comment (ArthurO2002) [inline src/pages/profile/help/index.tsx:35]: Move to translations

File: src/pages/profile/help/index.tsx [hunk 3]
```diff
@@ -1,22 +1,83 @@
...
+              />
+            )}
+          />
+        </div>
+        <div className="w-full">
+          <Button
+            size="lg"
+            type="submit"
+          >
+            Submit
```
Comment (ArthurO2002) [inline src/pages/profile/help/index.tsx:74]: Move to translations

File: src/pages/profile/help/index.tsx [hunk 4]
```diff
@@ -1,22 +1,83 @@
...
+          />
+        </div>
+        <div className="mb-5">
+          <Controller
+            control={control}
+            name="problem"
+            render={({ field, fieldState }) => (
+              <Input
+                {...field}
+                placeholder="Type your problem here..."
```
Comment (ArthurO2002) [inline src/pages/profile/help/index.tsx:62]: Translation

File: src/redux/slices/support/middleware.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+const { setError, setHelpLoading } = slice.actions
+
+const fetchHelpMessage = (message: IHelpDataProps) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setHelpLoading(true))
+
+    await API.support.sendHelpData(message)
+    dispatch(setError(null))
+    dispatch(setHelpLoading(false))
```
Comment (ArthurO2002) [inline src/redux/slices/support/middleware.ts:16]: No need as we have this in finaly

File: src/validation/support/help.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { object, string } from 'yup'
+
+export const helpValidationSchema = object({
+  headline: string().required('Headline is required'),
+  problem: string().min(5).required('Problem is required'),
```
Comment (ArthurO2002) [inline src/validation/support/help.ts:5]: Use transnlations

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/types/reduxTypes/supportStateTypes.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export interface ISupportProps {
+  support: ISupportPropsList
+}
+
+export interface ISupportPropsList {
+  error: null | string
```
[the remark this answers is not in the document]
Comment (alexkocharyan01) [reply src/types/reduxTypes/supportStateTypes.ts]: done

File: src/redux/slices/support/middleware.ts (hunk 5 above)
Comment (alexkocharyan01) [reply src/redux/slices/support/middleware.ts]: done

File: src/pages/profile/help/index.tsx (hunk 3 above)
Comment (alexkocharyan01) [reply src/pages/profile/help/index.tsx]: done

File: src/pages/profile/help/index.tsx (hunk 1 above)
Comment (alexkocharyan01) [reply src/pages/profile/help/index.tsx]: done

File: src/pages/profile/help/index.tsx [hunk 8]
```diff
@@ -1,22 +1,94 @@
...
+                error={fieldState.error ? t(Translation.HELP_HEADLINE_REQUIRED) : null}
+              />
+            )}
+          />
+        </div>
+        <div className="mb-5">
+          <Controller
+            control={control}
+            name="problem"
+            defaultValue=""
```
Comment (ArthurO2002) [inline src/pages/profile/help/index.tsx:69]: remove

File: src/redux/slices/support/middleware.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import { IError } from '@axios/authentication/authManagerTypes'
+import { AppDispatch } from '@redux/store'
+
+import slice from './slice'
+
+const { setError, setHelpLoading, setHelpStatus } = slice.actions
+
+const fetchHelpMessage = (message: IHelpDataProps) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setHelpLoading(true))
```
Comment (ArthurO2002) [inline src/redux/slices/support/middleware.ts:12]:
```suggestion

const createHelpMessage = (message: IHelpDataProps) => async (dispatch: AppDispatch) => {
  try {
    dispatch(setHelpLoading(true))
```

File: src/redux/slices/support/middleware.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+    const response = await API.support.sendHelpData(message)
+
+    dispatch(setHelpStatus(response.status))
```
Comment (ArthurO2002) [inline src/redux/slices/support/middleware.ts:16]: This should be boolean and in success case should be true

File: src/validation/support/help.ts (hunk 6 above)
Comment (ArthurO2002) [inline src/validation/support/help.ts:5]: We should put translation key here

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/profile/help/index.tsx [hunk 11]
```diff
@@ -1,22 +1,94 @@
...
+              <Input
+                {...field}
+                placeholder={`${t(Translation.HELP_HEADLINE)}`}
+                error={fieldState.error ? t(Translation.HELP_HEADLINE_REQUIRED) : null}
+              />
+            )}
+          />
+        </div>
+        <div className="mb-5">
+          <Controller
```
Comment (ArthurO2002) [inline src/pages/profile/help/index.tsx:66]: Use hook`useController` instead of Controller wrapper

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/profile/help/index.tsx (hunk 8 above)
Comment (alexkocharyan01) [reply src/pages/profile/help/index.tsx]: done

File: src/redux/slices/support/middleware.ts (hunk 9 above)
Comment (alexkocharyan01) [reply src/redux/slices/support/middleware.ts]: done

File: src/redux/slices/support/middleware.ts (hunk 10 above)
Comment (alexkocharyan01) [reply src/redux/slices/support/middleware.ts]: done

File: src/pages/profile/help/index.tsx (hunk 11 above)
Comment (alexkocharyan01) [reply src/pages/profile/help/index.tsx]: done

**Review by ArthurO2002 — APPROVED** [review]

File: src/ui-components/Input/Input.tsx [hunk 12]
```diff
@@ -23,87 +24,97 @@ export interface InputTypeCheckbox {
...
+          )}
+        </div>
+        {error ? <p className="mt-2.5 text-xs text-red">{error}</p> : null}
+      </>
+    )
+  }
+)
 
 export const InputCheckbox = React.forwardRef(
   (props: InputTypeCheckbox, ref: LegacyRef<HTMLInputElement>) => {
```
Comment (rikarsen) [inline src/ui-components/Input/Input.tsx:120]: you can destruct props like we did above inside main Input component

**Pull request merged**